docs/krp.md

```diff
@@ -0,0 +1,117 @@
+                - backupName: custom
+                    includedResources: ["sample1.myapp.mycompany.com", "sample2.myapp.mycompany.com", "sample3.myapp.mycompany.com"]
+                    # labelSelector: "" # intentionally omitted - don't require label match
+                    # includeClusterResources: false # by default
+                - backupName: deployments
+                    includedResources: ["Deployment"]
+                - backupName: everything
+                    includeClusterResources: true
+                    excludedResources: ["ConfigMap", "Secret", "Deployment", "sample1.myapp.mycompany.com", "sample2.myapp.mycompany.com", "sample3.myapp.mycompany.com"]  # don't restore resources we've already restored
+```
+
+## Explanation of the Capture and Recovery Specifications in the VRG
+
+The scope of a VRG disaster protection is a single Kubernetes namespace.  The
+VRG protects persistent volumes associated with the namespace and optionally
+protects Kubernetes resources in the namespace.  This documentation covers the
+product previews for protecting Kuberentes resources so does not explain the
+persistent volume disaster protection.
+
+The VRG enables Kubernetes resources to be catpured(backed up) and recovered as
+part of disaster protection.  This is accomplished through the
+kubeObjectProtection section of the VRG spec.  If kubeObjectProtection is not
+included in a VRG, then Kubernetes resources are not protected as part of the
+VRG disaster protection.
+
+The kubeObjectProtection section contains two sub-sections, captureOrder and
+recoverOrder.  This captureOrder section provides instructions on how to backup
+a namespaces Kubernetes resources.  The recoverOrder section provides
+instructions on how to recover a namespaces Kubernetes resources after a
+disaster.  This implies that the backup and recover order can be different and
+don't need to include all the same resources.  Each of the captureOrder and
+recoverOrder sections contain a list of resource instructions.  Each item in the
+list is acted upon even if it duplicates work done by other items in the list.
+So care should be taken to avoid duplication within either the backup or recover
+lists so that the best RPO and RTO are achieved.  The list items must meet the
+following requirements.  If the requirements are not met, then the operation of
+the lists is undefined.
+
+1. The name of each item in the captureOrder list must be unique
+1. The backupName of each item in the recoverOrder list much match a name in the
+ recoverOrder list
+1. A labelSelector in a list item only applies to that item in the list
+1. If a list item contains multiple labelSelectors then any resource that
+ matches either label selector is operated upon
+1. includeClusterResources in a list item only applies to that item in the list
+1. Each list item can contain either an includedResources section or an
+ excludedResources section, but not both
```

Comment:
I'm not sure this is true.  I think both can be specified.  But the only case that comes to mind for it is including resources matching a substring (e.g., *.apps) and then excluding some of those matches (e.g., replicasets.apps).  I haven't tested this and don't know if velero supports substring matching.

Reply:
What the code does and what we want users to know is different in this case.  So I'm documenting what we want users to know.